File: problems/61. Rotate List/main.py
class Solution(object):
    def rotateRight(self, head, k):
        """
        :type head: Optional[ListNode]
        :type k: int
        :rtype: Optional[ListNode]
        """
        if not head or not head.next or k == 0:
            return head

        tail = head
        counter = 1
        while tail.next:
            tail = tail.next
            counter += 1

        k = k % counter
        if k == 0:
            return head

        tail.next = head

        new_tail = head
        for _ in range(counter - k - 1):
            new_tail = new_tail.next

        new_head = new_tail.next
        new_tail.next = None

        return new_head

File: problems/61. Rotate List/main.go
func rotateRight(head *ListNode, k int) *ListNode {
	if head == nil || head.Next == nil || k == 0 {
		return head
	}

	tail := head
	counter := 1
	for tail.Next != nil {
		tail = tail.Next
		counter += 1
	}

	k %= counter
	if k == 0 {
		return head
	}

	tail.Next = head

	new_tail := head
	for i := 0; i < counter-k-1; i++ {
		new_tail = new_tail.Next
	}

	new_head := new_tail.Next
	new_tail.Next = nil

	return new_head
}
